acceptance-tests/src/acceptance-test/java/tech/pegasys/teku/test/acceptance/ValidatorKeysAcceptanceTest.java

```diff
@@ -0,0 +1,154 @@
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.test.acceptance;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.Iterator;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.io.TempDir;
+import tech.pegasys.teku.bls.BLSPublicKey;
+import tech.pegasys.teku.provider.JsonProvider;
+import tech.pegasys.teku.test.acceptance.dsl.AcceptanceTestBase;
+import tech.pegasys.teku.test.acceptance.dsl.BesuNode;
+import tech.pegasys.teku.test.acceptance.dsl.TekuNode;
+import tech.pegasys.teku.test.acceptance.dsl.TekuValidatorNode;
+import tech.pegasys.teku.test.acceptance.dsl.tools.deposits.ValidatorKeystores;
+
+public class ValidatorKeysAcceptanceTest extends AcceptanceTestBase {
+  private final JsonProvider jsonProvider = new JsonProvider();
+
+  @Test
+  void shouldMaintainValidatorsInMutableClient(@TempDir final Path tempDir) throws Exception {
+    final String networkName = "less-swift";
+    final BesuNode eth1Node = createBesuNode();
+    eth1Node.start();
+
+    final ValidatorKeystores validatorKeystores =
+        createTekuDepositSender(networkName).sendValidatorDeposits(eth1Node, 8);
+    final ValidatorKeystores extraKeys =
+        createTekuDepositSender(networkName).sendValidatorDeposits(eth1Node, 1);
+
+    final TekuNode beaconNode =
+        createTekuNode(config -> config.withNetwork(networkName).withDepositsFrom(eth1Node));
+
+    final TekuValidatorNode validatorClient =
+        createValidatorNode(
+                config ->
+                    config
+                        .withNetwork(networkName)
+                        .withValidatorApiEnabled()
+                        .withInteropModeDisabled()
+                        .withBeaconNode(beaconNode))
+            .withMutableValidatorKeystores(validatorKeystores);
+
+    beaconNode.start();
+    validatorClient.start();
+
+    getValidatorListing(validatorClient, 0);
+
+    addValidatorsAndExpect(validatorClient, validatorKeystores, tempDir, "imported");
+
+    validatorClient.waitForLogMessageContaining("Added validator");
+    validatorClient.waitForLogMessageContaining("Published block");
+    getValidatorListing(validatorClient, 8);
+
+    // second add attempt would be duplicates
+    addValidatorsAndExpect(validatorClient, validatorKeystores, tempDir, "duplicate");
+
+    // a random key won't be found, remove should give not_found
+    removeValidatorAndCheckStatus(validatorClient, extraKeys.getPublicKeys().get(0), "not_found");
+
+    // wait for epoch 1 to ensure the validator has attested, which will give slashing protection
+    // data
+    beaconNode.waitForLogMessageContaining("Epoch Event *** Epoch: 2");
```

Comment:
We should have a better way of waiting for epochs rather than just looking at log messages.  It could be taken from metrics.  Otherwise if we change the layout of Epoch Event logs we have to go update a bunch of ATs.